PhysicsTools/PatAlgos/plugins/PATIsolatedTrackProducer.cc

```diff
@@ -374,6 +385,20 @@ void pat::PATIsolatedTrackProducer::produce(edm::Event& iEvent, const edm::Event
 
     int trackQuality = gentk.qualityMask();
 
+    // get combined probQ and probXY
+    float probQonTrack = 0, probXYonTrack = 0, probQonTrackNoLayer1 = 0, probXYonTrackNoLayer1 = 0;
+    if (!gt2siPixelTrackProbQXY_.isUninitialized() && gt2siPixelTrackProbQXY->contains(tkref.id())) {
+      const reco::SiPixelTrackProbQXY siPixelTrackProbQXY = (*gt2siPixelTrackProbQXY)[tkref];
+      probQonTrack = siPixelTrackProbQXY.probQ();
+      probXYonTrack = siPixelTrackProbQXY.probXY();
+    }
+
+    if (!gt2siPixelTrackProbQXYNoLayer1_.isUninitialized() && gt2siPixelTrackProbQXYNoLayer1->contains(tkref.id())) {
+      const reco::SiPixelTrackProbQXY siPixelTrackProbQXYNoLayer1 = (*gt2siPixelTrackProbQXYNoLayer1)[tkref];
+      probQonTrackNoLayer1 = siPixelTrackProbQXYNoLayer1.probQ();
+      probXYonTrackNoLayer1 = siPixelTrackProbQXYNoLayer1.probXY();
```

Comment:
@slava77 was I supposed to multiply these with 255 for the `uint8_t` trick to work?

Reply:
well, there is no magic; but I would make the conversion in DataFormats/PatCandidates/interface/IsolatedTrack.h constructor(s). The user code is not supposed to know how the data is stored (encapsulation).

Reply:
I did 
```
   uint8_t probQonTrack_, probXYonTrack_, probQonTrackNoLayer1_, probXYonTrackNoLayer1_;
``` 
in the header already
I also have
```
    float probQonTrackNoLayer1() const { return probQonTrackNoLayer1_ / 255.; }
    float probXYonTrackNoLayer1() const { return probXYonTrackNoLayer1_ / 255.; }
```
but then where is the point you suggest to do the multiplication with the 255.?

Reply:
>  in IsolatedTrack.h constructor(s)